src/test/crimson/test_interruptible_future.cc

```diff
@@ -148,20 +150,97 @@ TEST_F(seastar_test_suite_t, loops)
 	    using namespace std::chrono_literals;
 	    return interruptor::make_interruptible(seastar::now()).then_interruptible([&vec] {
 	      return interruptor::do_for_each(std::begin(vec), std::end(vec), [](int) {
-		ceph_assert(::crimson::interruptible::interrupt_cond<TestInterruptCondition>);
-		return ::crimson::errorator<::crimson::ct_error::enoent>::make_ready_future<>();
+		ceph_assert(interruptible::interrupt_cond<TestInterruptCondition>);
+		return errorator<ct_error::enoent>::make_ready_future<>();
 	      }).safe_then_interruptible([] {
-		ceph_assert(::crimson::interruptible::interrupt_cond<TestInterruptCondition>);
+		ceph_assert(interruptible::interrupt_cond<TestInterruptCondition>);
 		return seastar::now();
-	      }, ::crimson::errorator<::crimson::ct_error::enoent>::all_same_way([] {
-		ceph_assert(::crimson::interruptible::interrupt_cond<TestInterruptCondition>);
+	      }, errorator<ct_error::enoent>::all_same_way([] {
+		ceph_assert(interruptible::interrupt_cond<TestInterruptCondition>);
 	      }));
 	    });
 	  });
 	}).then_interruptible([] {
-	  ceph_assert(::crimson::interruptible::interrupt_cond<TestInterruptCondition>);
+	  ceph_assert(interruptible::interrupt_cond<TestInterruptCondition>);
 	  return seastar::now();
 	});
       }, [](std::exception_ptr) {}, false).get0();
   });
 }
+
+using base_intr = interruptible::interruptor<TestInterruptCondition>;
+
+using base_ertr = errorator<ct_error::enoent, ct_error::eagain>;
+using base_iertr = interruptible::interruptible_errorator<
+  TestInterruptCondition,
+  base_ertr>;
+
+using base2_ertr = base_ertr::extend<ct_error::input_output_error>;
+using base2_iertr = interruptible::interruptible_errorator<
+  TestInterruptCondition,
+  base2_ertr>;
+
+template <typename F>
+auto with_intr(F &&f) {
+  return base_intr::with_interruption_to_error<ct_error::eagain>(
+    std::forward<F>(f),
+    TestInterruptCondition(false));
+}
+
+TEST_F(seastar_test_suite_t, errorated)
+{
+  run_async([] {
+    base_ertr::future<> ret = with_intr(
+      []() {
+	return base_iertr::now();
+      }
+    );
+    ret.unsafe_get0();
+  });
+}
+
+TEST_F(seastar_test_suite_t, errorated_value)
+{
+  run_async([] {
+    base_ertr::future<int> ret = with_intr(
+      []() {
+	return base_iertr::make_ready_future<int>(
+	  1
+	);
+      });
+    EXPECT_EQ(ret.unsafe_get0(), 1);
+  });
+}
+
+TEST_F(seastar_test_suite_t, expand_errorated_value)
+{
+  run_async([] {
+    base2_ertr::future<> ret = with_intr(
+      []() {
+	return base_iertr::make_ready_future<int>(
+	  1
+	).si_then([](auto) {
+	  return base2_iertr::make_ready_future<>();
+	});
+      });
+    ret.unsafe_get0();
+  });
+}
+
+TEST_F(seastar_test_suite_t, handle_error)
```

Comment:
this is the test which can keep GCC-9 busy forever.

Reply:
Interesting.  Thank!  I think I'll commit it out for now?